src/router/bundle/bundleParser.ts

```diff
@@ -353,9 +358,14 @@ export default class BundleParser {
                 let rootUrl = fullUrlMatch[1];
                 // If the reference doesn't have a urlRoot, check if the entry's fullUrl has a urlRoot
                 if (rootUrl === undefined) {
-                    if (entry.fullUrl && entry.fullUrl.match(captureFullUrlParts)) {
+                    if (entry.fullUrl) {
```

Comment:
Thoughts on avoiding the deeply nested if statement by breaking the code into two if statements
```
if (rootUrl === undefined && entry.fullUrl && entry.fullUrl.length > MAX_BUNDLE_ENTRY_URL_LENGTH) {  // throw error} 
elseif (rootUrl === undefined && enty.fullUrl) {
 rootUrl = entry.fullUrl.match(captureFullUrlParts)
                            ? entry.fullUrl.match(captureFullUrlParts)[1]
                            : undefined;
}
```

Reply:
Seems like CodeQL will still flag it if the two clause are separated completely. I took a middle path of organizing it to two layer of if statement.